ext/nio4r/org/nio4r/Selector.java

```diff
@@ -136,7 +137,14 @@ public IRubyObject register(ThreadContext context, IRubyObject io, IRubyObject i
     @JRubyMethod
     public IRubyObject deregister(ThreadContext context, IRubyObject io) {
         Ruby runtime = context.getRuntime();
-        Channel rawChannel = RubyIO.convertToIO(context, io).getChannel();
+        Channel rawChannel;
+
+        try {
+            rawChannel = RubyIO.convertToIO(context, io).getChannel();
+        } catch(IOError e) {
+            // The IO object we're trying to deregister has been closed
```

Comment:
I'm curious if there are other possible exceptions here besides the object being closed which shouldn't get swallowed.

Reply:
I think we'd be safer if we checked for closed IO rather than catching an exception. Give me 5m to demo this.